bin/oci-kvm

```diff
@@ -142,6 +181,85 @@ def destroy_vm(args):
     return oci_utils.kvm.virt.destroy(args.domain, args.destroy_disks)
 
 
+def _delete_network_vm(args):
+    """
+    Deletes a libvirt network
+
+    Parameters
+    -----------
+    args :
+        dict as the one returned by argparse.ArgumentParser().parse_args()
+        expected key :
+            net : the IP of the vNIC to be used.
+    Returns
+    -------
+        int
+            The return value provided by the kvm storage pool create
+            call, 0 on success, 1 otherwise.
+    """
+    libvirtConn = libvirt.openReadOnly(None)
+    if libvirtConn is None:
+        print >> sys.stderr, 'Cannot contact hypervisor'
+        return 1
+    net = None
+    try:
+        net = libvirtConn.networkLookupByName(args.network_name)
+    except libvirt.libvirtError:
+        print >> sys.stderr, 'Cannot find network named [%s]' % args.network_name
+        return 1
+    print 'Network found:\n'
+    print xml.dom.minidom.parseString(net.XMLDesc()).toprettyxml(indent=" ", newl='')
+    print ''
+    if raw_input('Really destroy this network ?') == 'y':
+        return oci_utils.kvm.virt.delete_virtual_network(network_name=args.network_name)
+    return 1
+
+
+def _create_network_vm(args):
+    """
+    Creates a libvirt network
+
+    Parameters
+    -----------
+    args :
+        dict as the one returned by argparse.ArgumentParser().parse_args()
+
+    Returns
+    -------
+        int
+            The return value provided by the kvm storage pool create
+            call, 0 on success, 1 otherwise.
+    """
+    # check network  name unicity
+    conn = libvirt.openReadOnly(None)
+    _vnets = []
+    if conn:
+        _vnets = [n.name() for n in conn.listAllNetworks() if n.name() == args.network_name]
+        conn.close()
+    else:
+        print >> sys.stderr, 'Cannot contact hypervisor'
+        return 1
+    if len(_vnets) != 0:
+        print >> sys.stderr, "Network with name [%s] already exists" % args.network_name
+        return 1
+
+    script = oci_utils.kvm.virt.create_virtual_network(network=args.net,
+                                                       network_name=args.network_name,
+                                                       ip_first=args.ip_first,
+                                                       ip_prefix=args.ip_prefix,
+                                                       ip_start=args.ip_start,
+                                                       ip_end=args.ip_end,
+                                                       vcn_cidr=args.vcn_cidr)
+    if script:
+        print 'Use the following commands to build a script to make the network created accross reboots'
```

Comment:
the Libvirt network is persisted .  All the routing is not. As explained in the document 
user must create  script to create the routing . 
To persiste routing we must add configuration files in /etc/sysconfig/network-scripts 
but even then there is some chicken and egg issue because virtual network related device are not there when such script are run at boot.